pkg/sql/schemachanger/scplan/internal/scstage/build.go

```diff
@@ -320,12 +320,29 @@ func (bc buildContext) makeStageBuilderForType(bs buildState, opType scop.Type)
 		// Determine whether there are any backfill or validation operations
 		// remaining which should prevent the scheduling of any non-revertible
 		// operations. This information will be used when building the current
-		// set of targets below.
+		// set of targets below. We will do this over the set of generated operations,
+		// since some of these may be no-oped for newly created tables.
 		for _, n := range nodes {
-			if oe, ok := bc.g.GetOpEdgeFrom(n); ok && oe.CanFail() {
-				sb.anyRemainingOpsCanFail = true
+			if sb.anyRemainingOpsCanFail {
 				break
 			}
+			s := n
+			for {
+				if s.CurrentStatus == s.TargetStatus {
+					break
+				}
+				if oe, ok := bc.g.GetOpEdgeFrom(s); ok {
+					for _, op := range oe.Op() {
+						if op.Type() != scop.MutationType {
+							sb.anyRemainingOpsCanFail = true
+							break
+						}
+					}
+					s = oe.To()
+				} else {
+					break
+				}
+			}
```

Comment:
This change makes sense but I'm a bit concerned about the added computational complexity that this loop adds. Can we pre-compute the `anyRemainingOpsCanFail` property for each node instead? Define `anyRemainingOpsCanFail := make(map[*screl.Node]bool), then for each target, start with the sink node and do a depth-first traversal going backwards along each op edge, if you've verified `oe.CanFail()` at least once in the path to the sink node then set `anyRemainingOpsCanFail[oe.From()] = true` and vice-versa.